Stage menu redux

tiennou opened the pull request:
The promised re-jig of #30 after much cleaning up has happened. This is based on #50.

Long story short :
- `PBGitIndexController` is no more, because having that weird not-a-controller object around felt too clunky (just look at the amount of calls that would need `commitController`) and didn't work well with the responder chain.
- `PBOpenFiles` has been folded inside `PBGitWindowController`.

I'd be tempted to rename `PBGitCommitController` to `PBStageViewController` for clarity reasons but meh.

I haven't tested all things yet, so maybe I broke some menu items. Also, I don't know how I feel about the Stage menu item changing names all the time. Given that I'm a heavy user of macOS's shortcut relabeling function, this is a big impediment to it working correctly, so I might revert that behavior.

tiennou commented:
Looks like we have 2 "Reveal in Finder" menu items now 😆.

ssp commented on Classes/Controllers/PBGitCommitController.h:
```diff
@@ -9,43 +9,16 @@
-@property(assign) BOOL stashKeepIndex;
+@interface PBGitCommitController : PBViewController
 
 - (IBAction) refresh:(id) sender;
```
Just wondering: What qualifies IBActions to appear in the Interface?

My POV would be that all IBActions should be part of the interface as their role is to be called from the outside.

tiennou replied:
I can't say any more. Given that the primary caller of IBActions is NIB files, and those don't really care about public/private (since everything is really public).

I understand your view, but I can give a counter-example : most of `PBGitCommitController` IBActions now depend on having the correct sender passed to them (IOW, a menu item, which ~has its menu delegate set to~ I can't even explain anymore, look around `-validateMenuItem:` 😞). I'm not really satisfied with that, but for the time-being it works (and I went through so much hidden hoops that I felt this was the best solution). I should have left a FIXME in here.

ssp commented:
Thanks for taking on the somewhat confusing controller structure @tiennou !

I tried around a litte and got the impression that some of the logic for handling Discard vs Trash is not present in the Main menu now (create a new file and you’ll get a discard menu item that doesn’t seem to work). I added a note to the code where I think things need to be changed.

Regarding the two Reveal In Finder Menu items: if the menu item only reveals the currently selected files from the index (as opposed to the repository itself – which I find perfectly acceptable), I think this menu item should not be in the Repository menu. Likewise for the Open Files menu item.

With the general size of these PRs I am pretty sure I didn’t manage to understand it fully, but for the sake of progress, I suggest we merge them in, so we can start building on the improved structure.

tiennou commented on Classes/Controllers/PBGitCommitController.m:
```diff
@@ -262,9 +262,20 @@ - (IBAction) forceCommit:(id) sender
+{
+	NSParameterAssert(sender != nil);
+
+	if ([sender isKindOfClass:[NSMenuItem class]]) return nil;
```
It is. Will fix. I actually seen it not work but went on to fix other things 😆.

tiennou commented:
Rebased, with fixes and an additional commit (I found a method I forgot to delete).

> I think this menu item should not be in the Repository menu.

Yeah, I removed the new one in favor of this original one. I'd be tempted to leave it here for the time being, because I don't know where else it could go... It doesn't really make sense in the File menu, and it's not only used in the Stage view (the History treeview has a Open file context menu too).

tiennou commented:
Merging in. In case anything comes up feel free to PR. I'm doing it mostly for the sake of the 5 other branches I have on top of that one that I keep rebasing around 😆.
